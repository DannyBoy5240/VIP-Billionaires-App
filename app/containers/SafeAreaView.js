import React from 'react'
import { StyleSheet } from 'react-native'
import { SafeAreaView as SafeAreaContext } from 'react-native-safe-area-context'
import { withTheme } from '../theme'

const styles = StyleSheet.create({
  view: {
    flex: 1,
  },
})

const SafeAreaView = React.memo(
  ({ style, children, testID, theme, vertical = true, ...props }) => (
    <SafeAreaContext
      style={[styles.view, style]}
      edges={vertical ? ['right', 'left'] : undefined}
      testID={testID}
      {...props}>
      {children}
    </SafeAreaContext>
  ),
)

export default withTheme(SafeAreaView)
