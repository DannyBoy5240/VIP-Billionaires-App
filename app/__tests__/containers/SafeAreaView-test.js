import React from 'react';
import renderer from 'react-test-renderer';
import SafeAreaView from '../../containers/SafeAreaView';

test('renders correctly', () => {
  const tree = renderer.create(<SafeAreaView />).toJSON();
  expect(tree).toMatchSnapshot();
});