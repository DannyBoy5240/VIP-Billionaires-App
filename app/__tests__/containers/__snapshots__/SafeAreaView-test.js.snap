// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "right",
      "left",
    ]
  }
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
